File: java/src/main/java/com/dddeurope/niuz/hr/AuthorService.java
package com.dddeurope.niuz.hr;

import com.dddeurope.niuz.events.ContractSigned;
import com.dddeurope.niuz.events.EventPublisher;

public class AuthorService {
    private final EventPublisher publisher;

    public AuthorService(EventPublisher publisher) {
        this.publisher = publisher;
    }

    public void hire(String authorId, String authorName, String bankAccount, String contractType, int rate) {
        HireAuthor command = new HireAuthor(authorId, authorName, contractType, rate, bankAccount);
        publisher.publish(new ContractSigned(command.getAuthorId(), command.getContractType(), command.getRate(), command.getBankAccount(), command.getAuthorName()));
    }
}

File: dotnet/niuz.application/hr/AuthorService.cs
using niuz.application.events;

namespace niuz.application.hr
{
    public class AuthorService
    {
        private readonly IEventPublisher publisher;

        public AuthorService(IEventPublisher publisher)
        {
            this.publisher = publisher;
        }

        public void Hire(HireAuthor command)
        {
            publisher.Publish(new ContractSigned(command.AuthorId, command.AuthorName, command.ContractType, command.Rate, command.BankAccount));
        }
    }
}